libkbfs/key_bundle_cache.go

```diff
@@ -24,21 +25,27 @@ type keyBundleCacheKey struct {
 	isWriter    bool
 }
 
-// NewKeyBundleCacheStandard constructs a new KeyBundleCacheStandard with the given
-// cache capacity.
-func NewKeyBundleCacheStandard(capacity int) *KeyBundleCacheStandard {
-	head, err := lru.New(capacity)
-	if err != nil {
-		panic(err.Error())
-	}
-	return &KeyBundleCacheStandard{head}
+func (k keyBundleCacheKey) str() string {
+	return k.tlf.String() + k.bundleIDStr + strconv.FormatBool(k.isWriter)
+}
+
+// NewKeyBundleCacheLRU constructs a new KeyBundleCacheStandard with LRU
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheLRU(capacityBytes int) *KeyBundleCacheStandard {
+	return &KeyBundleCacheStandard{cache.NewLRUEvictedCache(capacityBytes)}
+}
+
+// NewKeyBundleCacheRandom constructs a new KeyBundleCacheStandard with random
+// eviction strategy. The capacity of the cache is set to capacityBytes bytes.
+func NewKeyBundleCacheRandom(capacityBytes int) *KeyBundleCacheStandard {
+	return &KeyBundleCacheStandard{cache.NewRandomEvictedCache(capacityBytes)}
 }
 
 // GetTLFReaderKeyBundle implements the KeyBundleCache interface for KeyBundleCacheStandard.
 func (k *KeyBundleCacheStandard) GetTLFReaderKeyBundle(
 	tlf tlf.ID, bundleID TLFReaderKeyBundleID) (*TLFReaderKeyBundleV3, error) {
-	cacheKey := keyBundleCacheKey{tlf, bundleID.String(), false}
-	if entry, ok := k.lru.Get(cacheKey); ok {
+	cacheKey := keyBundleCacheKey{tlf, bundleID.String(), false}.str()
```

Comment:
Ah, I guess I was thinking the key was a `Measurable` also, but it's not.  Why not?